### PR TITLE
(feat) O3-2115: Using name templates in O3

### PR DESCRIPTION
**Pull request by vasharma05**
# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [x] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [x] My work includes tests or is validated by existing tests.
- [x] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
This PR introduces the name templates in the O3 framework. To use the name templates to get the name of the patient, you can use either of the 2 options below.

1. `PatientName` component
```
import {PatientName} from '@openmrs/esm-framework';

...

return <PatientName patientUuid={patientUuid} />
```

2. `usePatientName`
```
import {usePatientName} from '@openmrs/esm-framework';

const { isLoadingName, patientName, nameTemplate} = usePatientName(patientUuid) 
```

> **Please note that the person and patient UUIDs are the same.**

## Screenshots
None

## Related Issue
https://openmrs.atlassian.net/browse/O3-2115

### CODE REVIEW

**Comment by brandones** [comment]
Should we call this `usePersonName` and `<PersonName>`? I assume this is not just for patients (indeed in this PR you use this to display the user name).

File: packages/framework/esm-styleguide/src/_overrides.scss [hunk 1]
```diff
@@ -106,6 +106,15 @@
...
 
 /* Tabs */
+
+.cds--tabs .cds--tabs__nav-item--selected {
+  border-bottom: 2px solid var(--brand-03);
+}
+.cds--tabs .cds--tabs__nav-link:active,
+.cds--tabs .cds--tabs__nav-link:focus {
+  outline: 2px solid var(--brand-03);
+}
```
Comment (brandones) [inline packages/framework/esm-styleguide/src/_overrides.scss:116]: What's this do? Can we get a screenshot or something?

**Comment by brandones** [comment]
I am arguing against making this change in https://openmrs.atlassian.net/browse/O3-2115; I think we should solve this on the back-end.

**Comment by ibacher** [comment]
I agree that this is something we should fix on the backend.

**Pull request closed without merging**